src/cairo_runner.rs

```diff
@@ -1878,4 +2034,211 @@ mod test {
         });
     }
+
+    #[test]
+    fn cairo_run_with_trace_file() {
+        let path = String::from("cairo_programs/fibonacci.json");
+        let program = fs::read_to_string(path).unwrap();
+        let mut runner = PyCairoRunner::new(
+            program,
+            Some("main".to_string()),
+            Some("small".to_string()),
+            false,
+        )
+        .unwrap();
+
+        let trace_path = temp_dir().join("fibonacci.trace");
+        let trace_path = trace_path.to_str().unwrap();
+
+        _ = fs::remove_file(trace_path);
+
+        runner
+            .cairo_run_py(false, Some(trace_path), None, None, None, None)
+            .expect("Call to PyCairoRunner::cairo_run_py() failed.");
+
+        // We simply check if file exists
+        assert!(fs::canonicalize(trace_path).is_ok());
+
+        _ = fs::remove_file(trace_path);
+    }
+
+    #[test]
+    fn cairo_run_with_nonexistent_trace_file() {
+        let path = String::from("cairo_programs/fibonacci.json");
+        let program = fs::read_to_string(path).unwrap();
+        let mut runner = PyCairoRunner::new(
+            program,
+            Some("main".to_string()),
+            Some("small".to_string()),
+            false,
+        )
+        .unwrap();
+
+        let trace_path = "cairo_programs";
+
+        let result = runner.cairo_run_py(false, Some(trace_path), None, None, None, None);
+
+        assert!(result.is_err());
+    }
+
+    #[test]
+    fn cairo_run_with_memory_file() {
+        let path = String::from("cairo_programs/fibonacci.json");
+        let program = fs::read_to_string(path).unwrap();
+        let mut runner = PyCairoRunner::new(
+            program,
+            Some("main".to_string()),
+            Some("small".to_string()),
+            false,
+        )
+        .unwrap();
+
+        println!("{:?}", temp_dir().join("fibonacci.memory"));
+        let memory_path = temp_dir().join("fibonacci.memory");
+        let memory_path = memory_path.to_str().unwrap();
+
+        _ = fs::remove_file(memory_path);
+
+        runner
+            .cairo_run_py(false, None, Some(memory_path), None, None, None)
+            .expect("Call to PyCairoRunner::cairo_run_py() failed.");
+
+        // We simply check if file exists
+        assert!(fs::canonicalize(memory_path).is_ok());
+
+        _ = fs::remove_file(memory_path);
+    }
+
+    #[test]
+    fn cairo_run_with_nonexistent_memory_file() {
+        let path = String::from("cairo_programs/fibonacci.json");
+        let program = fs::read_to_string(path).unwrap();
+        let mut runner = PyCairoRunner::new(
+            program,
+            Some("main".to_string()),
+            Some("small".to_string()),
+            false,
+        )
+        .unwrap();
+
+        let memory_path = "cairo_programs";
+
+        let result = runner.cairo_run_py(false, None, Some(memory_path), None, None, None);
+
+        assert!(result.is_err());
```

Comment:
Are we using is_err() and not something a little bit more specific for something in particular?

Reply:
Because i cannot find a way to do it without triggering a `Can not import module: starkware.cairo.lang.vm.vm_exceptions:` panic error. I already tried something like this:
```rust
assert!(result.is_err());
assert!(result
        .unwrap_err()
        .to_string()
        .contains(&RunnerError::WriteFail.to_string()));
```
but the call to `PyErr::to_string()` seems to trigger it. Maybe we could change the _kind_ of error returned by the method to `PyValueError` like in _ids.rs_?